documentation/asciidoc/topics/ref_hotrod_configuration_builder.adoc

```diff
@@ -0,0 +1,59 @@
+[id='hotrod_configuration-{context}']
+= Hot Rod Configuration API
+You can programmatically configure Hot Rod client connections with the `ConfigurationBuilder` interface.
+
+[discrete]
+== On {k8s}
+
+Hot Rod clients running on {k8s} can use the following configuration:
+
+[source,java,options="nowrap",subs=attributes+]
+----
+import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
+import org.infinispan.client.hotrod.configuration.SaslQop;
+import org.infinispan.client.hotrod.impl.ConfigurationProperties;
+...
+
+ConfigurationBuilder builder = new ConfigurationBuilder();
+      builder.addServer()
+               .host("$SERVICE_HOSTNAME")
+               .port(ConfigurationProperties.DEFAULT_HOTROD_PORT)
+             .security().authentication()
+               .username("username")
+               .password("password")
+               .realm("default")
+               .saslQop(SaslQop.AUTH)
+               .saslMechanism("SCRAM-SHA-512")
+             .ssl()
+               .sniHostName("$SERVICE_HOSTNAME")
+               .trustStorePath("/path/to/tls.crt");
```

Comment:
@ryanemerson @Crumby for Hot Rod clients in k8s to use encryption how do you set `trustStorePath`? can you point to the secret that holds tls.crt? 

does it depend on if the client is in the same namespace as the cluster?

Reply:
I believe you should just be able to set `.trustStorePath"/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt")` if the pod is on the same k8 cluster. It's been a while since I looked at k8 tls though.

@rigazilla @Crumby does that sound right?

Reply:
I've forgot to reply.
It should work, but I've never tried it and since we're doing TLS directly I would do some test before and maybe update docs in a separate PR.